File: program61.java
///////////////////////////////////////////////////////
//
//  File name :     program61.java
//  Descreption :   Accept Input and Calculate Summation of Factors and Display
//  Author :        Vaibhav
//  Date :          
//
///////////////////////////////////////////////////////


import java.util.*;

class Number
{
    public int SumFactors(int iNo)
    {
        int iCnt = 0;
        int iSum = 0;
        for(iCnt = 1;iCnt <= (iNo/2);iCnt++)
        {
            if((iNo%iCnt) == 0)
            {
                iSum = iSum + iCnt;
            }
        
        }
        return iSum;
    }
}

class program61
{
    public static void main(String A[])
    {
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("---- This Application is used to Display Summation of Factors of Number ----");

        System.out.println("Enter number: ");
        int iValue = sobj.nextInt();

        Number nobj = new Number();
        iRet = nobj.SumFactors(iValue) ;

        System.out.println("Summation of Factors is: "+iRet);

    }
}

File: Java/program61.c
///////////////////////////////////////////////////////
//
//  File name :     program61.c
//  Descreption :   Accept Input and Calculate Summation of Factors and Display
//  Author :        Vaibhav
//  Date :          20/05/2025
//
///////////////////////////////////////////////////////

#include<stdio.h>

int DisplaySumFactors(int iValue)
{
    int iCnt = 0;
    int iSum = 0;

    for(iCnt = 1;iCnt <= (iValue / 2); iCnt++)
    {
        if((iValue % iCnt) == 0)
        {
            iSum = iSum + iCnt;
        }
    }
    return iSum;
}

int main()
{
    int iNo = 0;
    int iRet = 0;

    printf("---- This Application is used to Display Summation of Factors of a Number----\n");

    printf("Enter Number: ");
    scanf("%d",&iNo);

    iRet = DisplaySumFactors(iNo);

    printf("Summation of Factors is : %d",iRet);

    return 0;
}

